core/third_party/Distributions/README.md

```diff
@@ -0,0 +1,10 @@
+# Distributions
+
+These functions were adapted from `Boost`, version 1.70.0 (released April 12th, 2019 06:04 GMT).
```

Comment:
Thanks for adding this README, it's really helpful. Now that I see how the code was incorporated, I'm a little concerned at how different it is from the source in `boost`. With the other third-party libraries, we've been very careful not to make modifications to the files. There are a couple reasons for the concern:
1. The refactors introduce a non-trivial risk of bugs. We don't have any unit tests for these refactored classes, making it possible for a subtle bug to slip in.
2. If we want to bump to a new version of the library (in order to pick up a bug fix or optimization), it is hard to understand how to 're-copy' the new code into our repository.

Would it be possible to copy `boost/random/uniform_real_distribution.hpp` and `boost/random/poisson_distribution.hpp` wholesale, keeping the same directory structure and copying over any needed dependencies as well?

As a last note, we also need to include the boost license as we are redistributing the code/ creating a derivative work.